test/unit/dialect/TestDialect.td

```diff
@@ -0,0 +1,73 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2023 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+
+include "llvm-dialects/Dialect/Dialect.td"
+
+def TestDialect : Dialect {
+  let name = "test";
+  let cppNamespace = "test";
+}
+
+class TestOp<string mnemonic_, list<Trait> traits_>
+    : Op<TestDialect, mnemonic_, traits_>;
+
+def DialectOp1 : TestOp<"dialect.op.1",
+                       []> {
+  let results = (outs);
+  let arguments = (ins);
+
+  let defaultBuilderHasExplicitResultType = true;
```

Comment:
That doesn't make a lot of sense when there's no result value in the first place :)

I think this test dialect needs at least two ops, one with and one without overloads.

Reply:
Hm, the `DialectOp4` is used to show the intended behavior when a function is overloaded. Here, the same function is used with an `i32` and a `float` argument. This can be seen in the `DialectOpOverloadTests` - but I can extend the testing, if you want.